torch/utils/_pytree.py

```diff
@@ -422,18 +450,34 @@ def _namedtuple_unflatten(values: Iterable[Any], context: Context) -> NamedTuple
 
 
 def _namedtuple_serialize(context: Context) -> DumpableContext:
-    json_namedtuple = {
-        "class_name": context.__name__,
-        "fields": context._fields,
-    }
-    return json_namedtuple
+    if context not in SUPPORTED_SERIALIZED_TYPES:
+        raise NotImplementedError(
+            f"Can't serialize TreeSpec of namedtuple class {context} because we "
+            "didn't register a serializated_type_name. Please register using "
+            "`_register_namedtuple`."
+        )
+
+    serialize_node_def = SUPPORTED_SERIALIZED_TYPES[context]
+    serialized_type_name = serialize_node_def.serialized_type_name
```

Comment:
Out of curious, why not automatically determine the type name via `f'{cls.__module__}.{cls.__qualname__}'` if the type is a `namedtuple`? This will not need to introduce a such new `_register_namedtuple` API.

Reply:
Ah this is because this is not necessarily FC safe -- if users change the location of the defined type then it'll break the deserialization since that module will no longer exist.